cpp/src/games/nim/main/UnitTests.cpp

```diff
@@ -0,0 +1,96 @@
+#include <gtest/gtest.h>
+#include <games/nim/Game.hpp>
+
+using Game = nim::Game;
+using State = Game::State;
+using StateHistory = Game::StateHistory;
+using PolicyTensor = Game::Types::PolicyTensor;
+using IO = Game::IO;
+using Rules = Game::Rules;
+using SymmetryGroup = groups::TrivialGroup;
+using GameResults = core::WinShareResults<Game::Constants::kNumPlayers>;
+
+TEST(NimGameTest, InitialState) {
+  StateHistory history;
+  history.initialize(Rules{});
+  State state = history.current();
+
+  EXPECT_EQ(Rules::get_current_player(state), 0);
+  EXPECT_EQ(state.stones_left, 21); // Assuming the game starts with 21 stones
+}
+
+TEST(NimGameTest, MakeMove) {
+  StateHistory history;
+  history.initialize(Rules{});
+  Rules::apply(history, 2);
+  State state = history.current();
+
+  EXPECT_EQ(state.stones_left, 18);
+  EXPECT_EQ(Rules::get_current_player(state), 1);
+}
+
+TEST(NimGameTest, Player0Wins) {
+  StateHistory history;
+  history.initialize(Rules{});
+  Rules::apply(history, 2); // Player 0
+  Rules::apply(history, 2); // Player 1
+  Rules::apply(history, 2); // Player 0
+  Rules::apply(history, 2); // Player 1
+  Rules::apply(history, 2); // Player 0
+  Rules::apply(history, 2); // Player 1
+  Rules::apply(history, 2); // Player 0
+
+  GameResults::Tensor outcome;
+  core::action_t last_action = 2;
+  bool isTerminal = Rules::is_terminal(history.current(), 1 - history.current().current_player,
```

Comment:
Everywhere else in the code base, we name variables with lower-case and underscores. Let's stay consistent with that.